packages/vite/src/node/config.ts

```diff
@@ -817,6 +817,13 @@ export async function loadConfigFromFile(
         isTS = true
       }
     }
+
+    if (!resolvedPath) {
+      const cjsConfigFile = path.resolve(configRoot, 'vite.config.cjs')
+      if (fs.existsSync(cjsConfigFile)) {
+        resolvedPath = cjsConfigFile
```

Comment:
This needs to set isESM = false since it would have been set to true in a type: "module" project

Reply:
> since it would have been set to true in a type: "module" project

Not a problem afaik, all `isESM` controls is whether rollup bundles the config to an ESM or CommonJS file, and whether it's loaded via ESM dynamic import or a custom require loader. Bundling to ESM and loading via native dynamic import (as currently happens) works in a type: "module" project, no matter if the config is authored in CJS (with default export) or not.

But I can change it of course, should work either way :)

Reply:
Let's set `isESM` to false, even if it currently works, it is less surprising when following the code. We can merge this feat in the 2.8 beta